### PR TITLE
Patch for linux

### PR DESCRIPTION
**Pull request by NicBOMB**

### CODE REVIEW

**Comment by TanninOne** [comment]
Like the change to esptk, this contains changes unnecessary for what you're trying to achieve.

File: index.cpp [hunk 1]
```diff
@@ -370,8 +371,8 @@ class BSArchive: public Napi::ObjectWrap<BSArchive> {
     }
   }
 private:
-  std::string m_Name;
   std::shared_ptr<BSA::Archive> m_Wrapped;
+  std::string m_Name;
```
Comment (ghost) [inline index.cpp:375]: Fixes a clang warning.

File: index.cpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #include "bsatk/src/bsaarchive.h"
 #include "string_cast.h"
 #include <vector>
+#include <thread>
```
Comment (ghost) [inline index.cpp:4]: Missing header.

**Comment by ghost** [comment]
> Like the change to esptk, this contains changes unnecessary for what you're trying to achieve.

No, it _really is necessary_.

Since the package.json clone command uses `git clone` and `exit /b 0` it will succeed in the initial clone then fail in all subsequent builds with `sh: line 1: exit: /b: numeric argument required`. Apart from fixing the windows exit option error, there is still the bigger issue; a `git clone` of a hard dependency is really not ideal.

`bsatk` should instead be a submodule of this repo. Excluding it from this repo _will_ generate tedious build errors and require constantly deleting the stale copy in the bsatk directory when trying to debug this. For example, the directory `GitHub/Nexus-Mods/Vortex/extensions/gamebryo-bsa-support/node_modules/bsatk/bsatk/` may need to be deleted for the top-level build to succeed during this module's `git clone` command.

In this case, `TanninOne/modorganizer-bsatk` is not itself a node package, so should be explicitly added as a git submodule, including all of its files, rather than cloning and setting bsatk in the `.gitignore`. When modorganizer-bsatk changes, those updates need to be tracked. That repo has been stable for 7 years, so it's not like changes to that one submodule would be frequent either.

I can't see a downside to including that _necessary_ change, but I've now removed the submodule inclusion method from this and all my other pull requests to Vortex's module dependencies just to get my fixes into vortex_devel instead. This is definitely one of the reasons the Vortex README still needs a disclaimer:
> The build tools are unfortunately **not particularly robust**, so the build may break for various reasons (i.e. network problems, **dependencies that changed remotely**, ...) and leave the checkout in an **inconsistent state**.

I want to contribute fixes for anyone building this, not just Linux users. I'll save that for another round of PRs instead. :shrug:

**Comment by TanninOne** [comment]
@NicBOMB sorry for not replying earlier. A submodule will not work for this, that was my first instinct as well, see https://github.com/Nexus-Mods/node-bsatk/commit/9eb261983607784a8d45565920c4057df7a05312
The problem is that, at least at the time, npm would not pull submodules when installing modules from github dependencies so while this module would build on its own, having it as a dependency in package.json would not. That's why it's explicitly being cloned (with depth 0 so we don't fetch the entire history)

**Pull request merged**